tasks/human_eval_025.rs

```diff
@@ -5,14 +5,377 @@ HumanEval/25
 /*
 ### VERUS BEGIN
 */
+use vstd::arithmetic::div_mod::*;
+use vstd::arithmetic::mul::*;
+use vstd::assert_by_contradiction;
+use vstd::calc;
 use vstd::prelude::*;
 
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+pub closed spec fn is_prime(n: nat) -> bool {
+    forall|i: nat| 1 < i < n ==> #[trigger] (n % i) != 0
+}
+
+// canonical definition of prime factoriztion
+pub closed spec fn is_prime_factorization(n: nat, factorization: Seq<nat>) -> bool {
+    // all factors are prime
+    &&& forall|i: nat|
```

Comment:
Since the next line checks `0 <= i`, you could make `i` an `int` and save a cast on line 25.

Reply:
done!